tutorials/common/tutorials.js

```diff
@@ -192,8 +192,8 @@ function process_block_debounce(selector, debounce) {
     processBlockInfo.lastelem = selector;
     if (!debounce) {
       process_block(selector);
+      return;
     }
```

Comment:
@manthey could you explain why this change? Looking at the code it is not exactly clear to me.

Reply:
When a text area is altered, we want to rerun the tutorial, but we want to debounce it so it doesn't rerun too often (to give you a chance to finish typing).  We trigger this on textarea `change`, `input`, and `propertychange` events (and resetting a field).  The specific bug that this fixes is when no editing has been done, and you click on a text area and just delete a character.  In this case, because the area had not been the most recently altered, it enters this code path.  However, since we are trying to debounce, instead of processing the block, it should fall through and set a timer to run after debouncing.  But, the `return` was in the wrong place, and it never set the timer.  If you clicked on another text area, or typed more, then this branch (from line 188) wasn't entered, and the tutorial would update.  The `return` should only apply if we aren't debouncing and are processing the block immediately.